stravalib/model.py

```diff
@@ -149,58 +194,128 @@ def naive_datetime(value: Optional[AllDateTypes]) -> Optional[datetime]:
+
+        Parameters
+        ----------
+        attribute_value_mapping : dict
+            A dictionary representing the serialized data.
+
+        Returns
+        -------
+        DeprecatedSerializableMixin
+            A new instance of the class created from the serialized data.
+
+        Deprecated
+        ----------
+        1.0.0
+            The `deserialize()` method is deprecated in favor of `parse_obj()` method.
+            For more details, refer to the Pydantic documentation:
+            https://docs.pydantic.dev/usage/models/#helper-functions
+
+
         """
-        warn_method_deprecation(
+        exc.warn_method_deprecation(
             cls,
             "deserialize()",
             "parse_obj()",
             "https://docs.pydantic.dev/usage/models/#helper-functions",
         )
         return cls.parse_obj(attribute_value_mapping)
 
-    def from_dict(self, attribute_value_mapping: Dict):
+    def from_dict(self, attribute_value_mapping: dict[str, str]) -> None:
         """
-        Deserializes v into self, resetting and ond/or overwriting existing fields
+        Deserializes v into self, resetting and ond/or overwriting existing
+        fields
+
+        Parameters
+        ----------
+        attribute_value_mapping : dict
+            A dictionary that will be deserialized into the parent object.
+
+        Deprecated
+        ----------
+        1.0.0
+            The `from_dict()` method is deprecated in favor of `parse_obj()` method.
+            For more details, refer to the Pydantic documentation:
+            https://docs.pydantic.dev/usage/models/#helper-functions
         """
-        warn_method_deprecation(
-            self.__class__.__name__,
+
+        exc.warn_method_deprecation(
+            self.__class__,
             "from_dict()",
             "parse_obj()",
             "https://docs.pydantic.dev/usage/models/#helper-functions",
         )
-        # Ugly hack is necessary because parse_obj does not behave in-place but returns a new object
-        self.__init__(**self.parse_obj(attribute_value_mapping).dict())
+        # Ugly hack is necessary because parse_obj does not behave in-place but
+        # returns a new object
+        self.__init__(**self.parse_obj(attribute_value_mapping).dict())  # type: ignore
 
-    def to_dict(self):
+    def to_dict(self) -> Dict[str, str]:
         """
         Returns a dict representation of self
+
+        Returns
+        -------
+        dict
+            A dictionary containing the data from the instance.
+
+        Deprecated
+        ----------
+            The `to_dict()` method is deprecated in favor of `dict()` method.
+            For more details, refer to the Pydantic documentation:
+            https://docs.pydantic.dev/1.10/usage/exporting_models/
         """
-        warn_method_deprecation(
-            self.__class__.__name__,
+        exc.warn_method_deprecation(
+            self.__class__,  # expects klass: type (not a str)
```

Comment:
The comment can probably be removed or at least `klass -> class`

Reply:
i had no idea where that comment came from but think it was something mypy spit out. deleted!